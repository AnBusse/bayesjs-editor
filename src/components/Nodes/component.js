import React from 'react';
import Node from 'components/Node';

const Nodes = ({
  nodes,
  onMouseDown,
  onDoubleClick,
  onStateDoubleClick,
  contextItems,
}) => nodes.map((node) => {
  const props = {
    ...node,
    onMouseDown: onMouseDown(node),
    onDoubleClick: onDoubleClick(node),
    onStateDoubleClick: onStateDoubleClick(node),
  };

  return <Node key={node.id} contextItems={contextItems} {...props} />;
});

export default Nodes;
