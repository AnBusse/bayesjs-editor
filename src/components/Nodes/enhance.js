import { compose, pure, withHandlers } from 'recompose';
import { isFunction } from 'lodash';

const enhance = compose(
  pure,
  withHandlers({
    onMouseDown: ({ onMouseDown }) => node => e =>
      isFunction(onMouseDown) && onMouseDown(node, e),
    onDoubleClick: ({ onDoubleClick }) => node => e =>
      isFunction(onDoubleClick) && onDoubleClick(node, e),
    onStateDoubleClick: ({ onStateDoubleClick }) => node => state => () =>
      isFunction(onStateDoubleClick) && onStateDoubleClick(node, state),
  }),
);

export default enhance;
