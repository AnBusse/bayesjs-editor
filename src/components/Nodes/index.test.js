import React from 'react';
import { shallow } from 'enzyme';
import Nodes from './component';

const defaultProps = {
  nodes: [{}, {}],
  onMouseDown: jest.fn(),
  onDoubleClick: jest.fn(),
  onStateDoubleClick: jest.fn(),
};

const shallowComponent = (props = {}) => {
  const compProps = { ...defaultProps, ...props };

  return shallow(<Nodes {...compProps} />);
};

describe('Nodes Component', () => {
  it('matches snapshot', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });
});
