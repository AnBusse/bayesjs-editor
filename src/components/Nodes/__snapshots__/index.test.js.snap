// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nodes Component matches snapshot 1`] = `
Array [
  <Node
    belief={null}
    description={null}
    id="node-1"
    isSelected={false}
    key="node-1"
    onDoubleClick={[MockFunction]}
    onMouseDown={[MockFunction]}
    onStateDoubleClick={[MockFunction]}
    position={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    results={Object {}}
    size={
      Object {
        "height": 100,
        "width": 100,
      }
    }
    states={null}
  />,
  <Node
    belief={null}
    description={null}
    id="node-2"
    isSelected={false}
    key="node-2"
    onDoubleClick={[MockFunction]}
    onMouseDown={[MockFunction]}
    onStateDoubleClick={[MockFunction]}
    position={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    results={Object {}}
    size={
      Object {
        "height": 100,
        "width": 100,
      }
    }
    states={null}
  />,
]
`;
